Objects/abstract.c

```diff
@@ -2435,10 +2436,8 @@ method_output_as_list(PyObject *o, PyObject *meth)
         PyThreadState *tstate = _PyThreadState_GET();
         if (_PyErr_ExceptionMatches(tstate, PyExc_TypeError)) {
             _PyErr_Format(tstate, PyExc_TypeError,
-                          "%.200s.%U() returned a non-iterable (type %.200s)",
```

Comment:
Here I strongly prefer the old message. Iterable is not a type, it's a category of types.

Reply:
what do you think about new message: `"%T.%U() must return an iterable, not %T"`

Reply:
I don't think new message adds something bad.  Why someone might think, that iterable is a type here?  (Another similar case, where new messages mention an iterator.)

Reply:
> Why someone might think, that iterable is a type here?

it's just that in the previous iteration the error message was the phrase "must return type ..." :)